### PR TITLE
call manager start() and stop() in app init

### PR DESCRIPTION
**Pull request by achou11**
Haven't tested the changes on a device yet.

the latest version of core introduced `start()` and `stop()` methods on MapeoManager, which are currently used to handle the media server. due to this, i believe we should reintroduce using the rnBridge app lifecycle hooks, which presents a couple of questions:

- Do we need more variants of a server status now? e.g. `stopping`, `stopped`
- Does the message port need to also be started/stopped in the rnBridge lifecycle hooks?
  - may need to introduce a `pause()` method on the MessagePortLike if so

### CODE REVIEW

File: src/backend/src/app.js [hunk 1]
```diff
@@ -53,6 +52,19 @@ export async function init({ version, rootKey }) {
     coreStorage: () => new RAM(),
   })
 
+  await manager.start()
+
+  rnBridge.app.on('pause', async (pauseLock) => {
+    log('App went into background')
+    await manager.stop()
+    pauseLock.release()
+  })
```
Comment (achou11) [inline src/backend/src/app.js:61]: does `messagePort` need to be stopped here too?

File: src/backend/src/app.js [hunk 2]
```diff
@@ -53,6 +52,19 @@ export async function init({ version, rootKey }) {
...
+
+  rnBridge.app.on('resume', () => {
+    log('App went into foreground')
+    manager.start()
+  })
```
Comment (achou11) [inline src/backend/src/app.js:66]: does messagePort need to be started here too?

**Comment by gmaclennan** [comment]
> * Do we need more variants of a server status now? e.g. `stopping`, `stopped`

No, because the "server" in this "server status" refers to the IPC server, not the HTTP server, and that is not stopping. The `getUrl` methods internally wait for the server to start before resolving, so whether the HTTP server is stopped should not be relevant to using the API. The edge case is a "stale" URL - e.g. the frontend has a URL from before app pause, and when the app resumes it tries to load that URL. If the port has changed then that will not work. We can discuss how to resolve that in a separate issue.

> * Does the message port need to also be started/stopped in the rnBridge lifecycle hooks?

No, the message port doesn't open any ports or use any resources, so there is no need to start or stop it.

>   * may need to introduce a `pause()` method on the MessagePortLike if so

No need I don't think.

File: src/backend/src/app.js [hunk 3]
```diff
@@ -53,6 +52,19 @@ export async function init({ version, rootKey }) {
     coreStorage: () => new RAM(),
   })
 
+  await manager.start()
```
Comment (gmaclennan) [inline src/backend/src/app.js:55]: No need to await this I don't think, getUrl methods will wait internally. We might want to catch errors on these start and stops, however the unhandled rejection handler will do that and probably suffice for now unless we have a different message we want to show to the user in this case.

File: src/backend/src/app.js (hunk 1 above)
Comment (gmaclennan) [reply src/backend/src/app.js:61]: no

File: src/backend/src/app.js (hunk 2 above)
Comment (gmaclennan) [reply src/backend/src/app.js:66]: no

**Review by gmaclennan — APPROVED** [review]
Looks good. One non-blocking perf issue - no need to await http server start before initializing the IPC server and starting the message port.

**Comment by achou11** [comment]
> No, because the "server" in this "server status" refers to the IPC server, not the HTTP server, and that is not stopping

ah that makes sense, thanks for the clarification!

File: src/backend/src/app.js (hunk 3 above)
Comment (gmaclennan) [inline src/backend/src/app.js:55]:
```suggestion
// Don't await, methods that use the server will await this internally
manager.start()
```

**Comment by gmaclennan** [comment]
@ErikSin this is also good to merge when you are ready.

**Pull request merged**